mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/Property.java

```diff
@@ -51,11 +53,11 @@ public class Property {
      * @param localName not null
      * @param value not null
      */
-    public Property(String namespace, String localName, String value) {
+    public Property(@NonNull String namespace, @NonNull String localName, @NonNull String value) {
```

Comment:
just curious. this constructor will throw an Exception when input is null or is it just a "description Annotation"?

Reply:
Annotation is descriptive, but I also added explicit checks inside via `Objects.requireNonNull()`